src/document/mod.rs

```diff
@@ -84,113 +116,241 @@ impl DocumentEncryptOpts {
     async fn document_update_bytes(
         &self,
         id: &DocumentId,
         new_document_data: &[u8],
     ) -> Result<DocumentEncryptResult>;
 
-    /// Decrypts the provided encrypted document and returns details about the document as well as its decrypted bytes.
+    /// Modifies or removes a document's name.
     ///
-    /// # Arguments
-    /// - `encrypted_document` - Bytes of encrypted document. Should be the same bytes returned from `document_encrypt`.
-    ///
-    /// # Returns
-    /// `Result<DocumentDecryptResult>` Includes metadata about the provided document as well as the decrypted document bytes.
-    async fn document_decrypt(&self, encrypted_document: &[u8]) -> Result<DocumentDecryptResult>;
-
-    /// Update a document name to a new value or clear its value.
+    /// Returns the updated metadata of the document.
     ///
     /// # Arguments
-    /// - `id` - ID of the document to update
-    /// - `name` - New name for the document. Provide a Some to update to a new name and a None to clear the name field.
+    /// - `id` - Unique ID of the document to update
+    /// - `name` - New name for the document. Provide a `Some` to update to a new name or a `None` to clear the name field.
     ///
-    /// # Returns
-    /// `Result<DocumentMetadataResult>` Metadata about the document that was updated.
+    /// # Examples
+    /// ```
+    /// # async fn run() -> Result<(), ironoxide::IronOxideErr> {
+    /// # use ironoxide::prelude::*;
+    /// # let sdk: IronOxide = unimplemented!();
+    /// # let document_id: DocumentId = unimplemented!();
+    /// use std::convert::TryFrom;
+    /// let new_name = DocumentName::try_from("updated")?;
+    /// let document_meta = sdk.document_update_name(&document_id, Some(&new_name)).await?;
+    /// # Ok(())
+    /// # }
+    /// ```
     async fn document_update_name(
         &self,
         id: &DocumentId,
         name: Option<&DocumentName>,
     ) -> Result<DocumentMetadataResult>;
 
-    /// Grant access to a document. Recipients of document access can be either users or groups.
+    /// Grants decryption access to a document for the provided users and/or groups.
+    ///
+    /// Returns lists of successful and failed grants.
     ///
     /// # Arguments
-    /// `document_id` - id of the document whose access is is being modified
-    /// `grant_list` - list of grants. Elements represent either a user or a group.
+    /// `document_id` - Unique ID of the document whose access is being modified.
+    /// `grant_list` - List of users and groups to grant access to.
     ///
-    /// # Returns
-    /// Outer result indicates that the request failed either on the client or that the server rejected
-    /// the whole request. If the outer result is `Ok` then each individual grant to a user/group
-    /// either succeeded or failed.
+    /// # Errors
+    /// Fails only if the calling user does not have access to the document. If the response is an `Ok`, then the resulting
```

Comment:
Given that the request to ironcore-id can also fail, this is a little simplified. It's maybe not worthwhile to try and enumerate all of the reasons it can fail. What if we said something like

```
This operation supports partial success. If any of the provided grants fail the `DocumentAccessResult` will indicate which grants succeeded and which failed, and it will provide explanations for each failure.
```

Reply:
I added the part about partial success, removed the over-simplification, and tweaked the second sentence